src/lib.rs

```diff
@@ -901,6 +955,23 @@ impl Url {
     }
 
     /// Change this URL’s fragment identifier.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use url::Url;
+    ///
+    /// let mut url = Url::parse("https://example.com/data.csv").unwrap();
+    /// {
+    ///     url.set_fragment(Some("cell=4,1-6,2"));
+    /// }
+    /// assert_eq!(url.fragment(), Some("cell=4,1-6,2"));
```

Comment:
Please add an assert on `url.as_str()` after each of these to show what the full url looks like.